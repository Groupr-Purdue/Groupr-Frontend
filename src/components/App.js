// @flow

import React, { PropTypes } from 'react';
import { observer, inject } from 'mobx-react';
import AppBar from 'material-ui/AppBar';
import { Row, Col, Grid } from 'react-flexbox-grid-aphrodite';
import GroupWork from 'material-ui/svg-icons/action/group-work';
import IconButton from 'material-ui/IconButton';
import { CubeGrid } from 'better-react-spinkit';

const App =
  (
    { children, routing, navbar }:
    { children: Array<React$Element>, routing: Object, navbar: Object }
  ): React$Element =>
    <Grid>
      <Row style={{ marginBottom: '15px' }}>
        <Col xs={12}>
          <AppBar
            key={'AppBar'}
            title={navbar.title}
            onClick={(): Void => routing.push('/courses')}
            iconElementRight={
              navbar.loading ?
                <Row style={{ height: 50, marginRight: 10 }} middle='xs'>
                  <CubeGrid color='white' size={30} />
                </Row> : null
              }
            iconElementLeft={
              <IconButton
                tooltip='Team up!'
                style={{ padding: 0 }}
                iconStyle={{ height: 42, width: 42 }}>
                <GroupWork />
              </IconButton>
            } />
        </Col>
      </Row>
      <Row>
        <Col xs={12}>
          {children}
        </Col>
      </Row>
    </Grid>;

App.propTypes = {
  children: PropTypes.element,
  routing: PropTypes.object,
  navbar: PropTypes.object,
};

export default inject('routing', 'navbar')(observer(App));
