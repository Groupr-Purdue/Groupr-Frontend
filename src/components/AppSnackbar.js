import React from 'react';
import snackbar from '~/store/snackbar';
import { observer } from 'mobx-react';
import { Snackbar } from 'material-ui';

const AppSnackbar = (): Element =>
  <Snackbar {...snackbar.state} />;

export default observer(AppSnackbar);
