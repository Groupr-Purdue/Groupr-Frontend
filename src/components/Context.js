import React from 'react';
import { Provider } from 'mobx-react';

// required by react-grid-layout
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import 'react-grid-layout/css/styles.css';
import 'react-resizable/css/styles.css';

import navbarStore from '../store/navbar';
import routingStore from '../store/router';

const stores = {
  routing: routingStore,
  navbar: navbarStore,
};

export default (props: { children : Object }) =>
  <MuiThemeProvider>
    <Provider {...stores}>
      { props.children }
    </Provider>
  </MuiThemeProvider>
;
