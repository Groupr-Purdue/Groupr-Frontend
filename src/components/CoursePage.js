// @flow

import React from 'react';
import { inject, observer } from 'mobx-react';
import {
  Paper,
  List,
  ListItem,
  Subheader,
  Divider,
} from 'material-ui';
import course from '~/store/course';
import router from '~/store/router';
import { Grid, Row, Col } from 'react-flexbox-grid-aphrodite';
import loadingWrapper from '~/util/loadingWrapper';

const handleClick =
  (): Void => router.push(`/users/${course.id}`);

const CoursePage = (): Element =>
  <div>
    <Paper>
      <Grid>
        <Row center='xs'>
          <Col xs={10} style={{ padding: '20px 0 20px 0' }}>
            <div>
              <h1 id='course_name'> {course.name} </h1>
              <h2 id='course_title'> {course.title} </h2>
            </div>
          </Col>
        </Row>

      </Grid>
      <Divider />
      <Grid>
        <Row>
          <Col xsOffset={1} xs={10}>
            <List>
              <Subheader> {course.name} Students </Subheader>
              { do {
                if (course.users.length)
                  course.users.map(
                    (user: Object, idx: number): React$Element =>
                      <ListItem
                        key={idx}
                        primaryText={`${user.first_name} ${user.last_name}`}
                        secondaryText={user.career_account}
                        onClick={handleClick} />);

                else <ListItem primaryText='No studuents in this course.' />;
              } }
            </List>
          </Col>
        </Row>
      </Grid>
    </Paper>
  </div>;

export default loadingWrapper(observer(CoursePage));
