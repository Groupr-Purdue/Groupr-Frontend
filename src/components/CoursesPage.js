// @flow

import React from 'react';
import { observer } from 'mobx-react';
import {
  Paper,
  List,
  ListItem,
  Divider,
} from 'material-ui';
import courses from '~/store/courses';
import { Row, Col } from 'react-flexbox-grid-aphrodite';
import router from '~/store/router';
import loadingWrapper from '~/util/loadingWrapper';

const handleClick =
  (course: Object): Function =>
    async (): Void => {
      await courses.register(course.id + 1);
      router.push(`/courses/${course.id + 1}`);
    };

const CoursesPage = (): Object =>
  <Row center='xs'>
    <Col xs={12} sm={6} md={6} lg={6}>
      <Paper>
        <List>
          <Row center='xs'><h3> Your Courses: Select a course to join. </h3></Row>
          <Divider style={{ marginBottom: '10px' }} />
          { do {
            if (courses.list.length)
              courses.list.map((course: Object, idx: number): Element =>
                <ListItem
                  key={idx}
                  id={`course_${idx}`}
                  primaryText={course.title}
                  secondaryText={course.name}
                  onClick={handleClick(course)} />);

            else
              <Row center='xs'>
                <h4> No Courses </h4>
              </Row>;
              // <LinearProgress mode='indeterminate'
              //   style={{ marginBottom: 30 }} />;
          } }
        </List>
      </Paper>
    </Col>
  </Row>;

export { CoursesPage as NakedCoursesPage };

export default loadingWrapper(observer(CoursesPage), courses.loading);
