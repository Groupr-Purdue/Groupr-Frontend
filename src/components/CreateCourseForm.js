import React from 'react';
import { observer } from 'mobx-react';
import {
  Paper,
  Divider,
  TextField,
  FloatingActionButton,
} from 'material-ui';
import ContentAdd from 'material-ui/svg-icons/content/add';
import router from '~/store/router';
import { Row, Col } from 'react-flexbox-grid-aphrodite';
import course from '~/store/courseForm';
import loadingWrapper from '~/util/loadingWrapper';

const handleSubmit = () => {
  course.submit().then(({ id }) => router.push(`/courses/${id}`));
};

const CreateCourseForm = (): Element =>
  <div>
    <Row center='xs'>
      <Col xs={12} sm={6} md={6} lg={6} >
        <Paper>
          <Row center='xs'>
            <Col xs={10} style={{ padding: '20px 0 20px 0' }}>
              <div>
                <h1 id='course_name'> Create Course Form </h1>
              </div>
            </Col>
          </Row>
          <Divider />
          <Row center='xs'>
            <Col xs={10}>
              <Row>
                <TextField fullWidth={true}
                  floatingLabelText='Course Title'
                  hintText='Intro to Smoke Testing'
                  onChange={(ev, value) => course.title = value} />
              </Row>
              <Row>
                <TextField fullWidth={true}
                  floatingLabelText='Course Name'
                  hintText='CS 420'
                  onChange={(ev, value) => course.name = value} />
              </Row>
              <Row start='xs'>
                <TextField fullWidth={true}
                  floatingLabelText='White List'
                  hintText='jshmoe, morecareeraccounts...'
                  multiLine={true}
                  onChange={(ev, value) => course.users = value} />
              </Row>
              <Row center='xs'>
                <FloatingActionButton
                  style={{ margin: '0 0 15px 0', transform: 'translate(0, 40px)' }}
                  onClick={handleSubmit}>
                  <ContentAdd />
                </FloatingActionButton>
              </Row>
            </Col>
          </Row>
        </Paper>
      </Col>
    </Row>
  </div>;

export default loadingWrapper(CreateCourseForm, course.loading);
