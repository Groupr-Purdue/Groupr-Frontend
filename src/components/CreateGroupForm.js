import React from 'react';
import {
  Paper,
  Divider,
  TextField,
  FloatingActionButton,
} from 'material-ui';
import ContentAdd from 'material-ui/svg-icons/content/add';
import router from '~/store/router';
import { Row, Col } from 'react-flexbox-grid-aphrodite';
import groupForm from '~/store/groupForm';
import loadingWrapper from '~/util/loadingWrapper';

export const handleSubmit =
  id =>
    () =>
      groupForm.submit(id)
      .then(() => router.push(`/courses/${id}`))
      .catch(() => groupForm.failLoading());

const CreateGroupForm = (props: Array<Object>): Element =>
  <div>
    <Row center='xs'>
      <Col xs={12} sm={6} md={6} lg={6} >
        <Paper>
          <Row center='xs'>
            <Col xs={10} style={{ padding: '20px 0 20px 0' }}>
              <div>
                <h1> Group Course Form </h1>
              </div>
            </Col>
          </Row>
          <Divider />
          <Row center='xs'>
            <Col xs={10}>
              <Row>
                <TextField fullWidth={true}
                  floatingLabelText='Group Name'
                  hintText='The Super Stars'
                  onChange={(ev, value) => groupForm.name = value} />
              </Row>
              <Row center='xs'>
                <FloatingActionButton
                  style={{ margin: '0 0 15px 0', transform: 'translate(0, 40px)' }}
                  onClick={handleSubmit(props[0].params.id)}>
                  <ContentAdd />
                </FloatingActionButton>
              </Row>
            </Col>
          </Row>
        </Paper>
      </Col>
    </Row>
  </div>;

export default loadingWrapper(CreateGroupForm, groupForm.loading);
