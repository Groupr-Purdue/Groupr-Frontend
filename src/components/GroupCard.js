import React from 'react';
import {
  Card,
  CardHeader,
  CardText,
  FlatButton,
} from 'material-ui';
import UserList from '~/components/UserList';
import ContentAdd from 'material-ui/svg-icons/content/add';
import ContentRemove from 'material-ui/svg-icons/content/remove';
import course from '~/store/course';
import user from '~/store/user';
import snackbar from '~/store/snackbar';
import { observer } from 'mobx-react';

type propType = { group: Object, style: Object };

const handleJoinGroup =
  id =>
    () =>
      course.joinGroup(id).then(snackbar.open);

const handleLeaveGroup =
  id =>
    () =>
      course.leaveGroup(id);

const GroupCard = ({ group, style = {} }: propType): Element =>
  <Card style={style}>
    <CardHeader
      title={group.name}
      actAsExpander={true}
      showExpandableButton={true} />
    <CardText expandable={true}>
      <UserList users={group.members} />
      { do {
        if (user.loggedIn && !group.members.find(member => member.id === user.id))
          <FlatButton
            primary={true}
            label='Join Group'
            icon={<ContentAdd />}
            onClick={handleJoinGroup(group.id)} />;
        else if (user.loggedIn)
          <FlatButton
            secondary={true}
            label='Leave Group'
            icon={<ContentRemove />}
            onClick={handleLeaveGroup(group.id)} />;
      } }
    </CardText>
  </Card>;

export default observer(GroupCard);
