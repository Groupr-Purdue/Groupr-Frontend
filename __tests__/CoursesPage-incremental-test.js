import React from 'react';
import { NakedCoursesPage as CoursesPage } from '../src/components/CoursesPage';
import Context from '../src/components/Context';
import { shallow } from 'enzyme';

test('jest CoursesPages snapshot test', () => {
  expect(shallow(<CoursesPage />)).toMatchSnapshot();
});
