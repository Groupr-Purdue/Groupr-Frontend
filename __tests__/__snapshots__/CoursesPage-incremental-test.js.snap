exports[`test jest CoursesPages snapshot test 1`] = `<CoursesPage />`;
